lib/media/track/remotetrackpublication.js

```diff
@@ -94,14 +98,26 @@ class RemoteTrackPublication extends TrackPublication {
         }
         this.emit(signaling.isEnabled ? 'trackEnabled' : 'trackDisabled');
       }
-      if (isSwitchedOff !== signaling.isSwitchedOff) {
-        this._log.debug(`${this.trackSid}: ${isSwitchedOff ? 'OFF' : 'ON'} => ${signaling.isSwitchedOff ? 'OFF' : 'ON'}`);
+      const newSwitchOffReason = signaling.switchOffReason || null;
+      if (isSwitchedOff !== signaling.isSwitchedOff || switchOffReason !== newSwitchOffReason) {
+        log.debug(`${this.trackSid}: ${isSwitchedOff ? 'OFF' : 'ON'} => ${signaling.isSwitchedOff ? 'OFF' : 'ON'}`);
+        log.debug(`${this.trackSid} off_reason: ${switchOffReason} => ${newSwitchOffReason}`);
         isSwitchedOff = signaling.isSwitchedOff;
+        switchOffReason = newSwitchOffReason;
+        if (this.track) {
+          this.track._setSwitchedOff(signaling.isSwitchedOff, switchOffReason);
+          this.emit(`trackSwitched${isSwitchedOff ? 'Off' : 'On'}`,  this.track, ...(isSwitchedOff ? [this.track.switchOffReason] : []));
```

Comment:
NIT. breaking strings make it harder to search for code references. 
```suggestion
          this.emit(isSwitchedOff ? 'trackSwitchedOff' : 'trackSwitchedOn',  this.track, ...(isSwitchedOff ? [this.track.switchOffReason] : []));
```